### PR TITLE
Added Priority in Broken Rule

### LINKED ISSUE #3224: Provide more details about the rule in the BrokenRule type
@rockfordlhotka Thank you.
I would like to make a couple of suggestions for the next version of Csla
1) We could add 'GetFirstBrokenRules()' in BrokenRulesCollection, this method will return 'first' broken rules for the object (or object graph).
2) Add Priority to BrokenRule.

Kind Regards

_Originally posted by @Chicagoan2016 in https://github.com/MarimerLLC/csla/discussions/3222#discussioncomment-4392507_

### PR DESCRIPTION
**Pull request by Freelancingonupwork**
-Updated  'GetFirstBrokenRules()' in BrokenRulesCollection, this method will return 'first' broken rules for the object (or object graph).
-Added Priority to `BrokenRule.cs`

Fixes #3224

### CODE REVIEW

File: Source/Csla/Rules/BrokenRule.cs [hunk 1]
```diff
@@ -28,8 +28,12 @@ public BrokenRule()
     private string _property;
     private RuleSeverity _severity;
     private string _originProperty;
+
```
Comment (StefanOssendorf) [inline Source/Csla/Rules/BrokenRule.cs:31]: Why empty lines here when all other fields are below each other?

File: Source/Csla/Rules/BrokenRule.cs [hunk 2]
```diff
@@ -117,8 +134,12 @@ protected override void OnGetState(SerializationInfo info, StateMode mode)
       info.AddValue("_property", _property);
       info.AddValue("_severity", (int)_severity);
       info.AddValue("_originProperty", _originProperty);
+
```
Comment (StefanOssendorf) [inline Source/Csla/Rules/BrokenRule.cs:137]: Why empty lines when all other code is below each other?

File: Source/Csla/Rules/BrokenRule.cs [hunk 3]
```diff
@@ -117,8 +134,12 @@ protected override void OnGetState(SerializationInfo info, StateMode mode)
...
+      info.AddValue("_priority", _priority);
+
       info.AddValue("_displayIndex", (int)_displayIndex);
 
+
```
Comment (StefanOssendorf) [inline Source/Csla/Rules/BrokenRule.cs:142]: Remove empty line

File: Source/Csla/Rules/BrokenRulesCollection.cs [hunk 4]
```diff
@@ -72,7 +72,7 @@ internal void ClearRules(Csla.Core.IPropertyInfo property)
       }
     }
     
-    internal void SetBrokenRules(List<RuleResult> results, string originPropertyName)
+    internal void SetBrokenRules(List<RuleResult> results, string originPropertyName, int Priority)
```
Comment (StefanOssendorf) [inline Source/Csla/Rules/BrokenRulesCollection.cs:75]: Parameters have to be camelCase. So it must be priority not Priority.

File: Source/Csla/Rules/BrokenRulesCollection.cs [hunk 5]
```diff
@@ -112,7 +112,11 @@ internal void SetBrokenRules(List<RuleResult> results, string originPropertyName
                        ? null : resultPrimaryProperty.Name,
             Severity = result.Severity,
             OriginProperty = originPropertyName,
+
```
Comment (StefanOssendorf) [inline Source/Csla/Rules/BrokenRulesCollection.cs:115]: Why empty lines here? (Like the others)

**Review by StefanOssendorf — CHANGES_REQUESTED** [review]

**Review by StefanOssendorf — APPROVED** [review]
@rockfordlhotka The changes are okay for me. But please double check if it's what was intended with the associated issue. Thanks

File: Source/Csla/Rules/BrokenRulesCollection.cs [hunk 6]
```diff
@@ -282,7 +283,7 @@ public BrokenRule GetFirstMessage(Core.IPropertyInfo property, RuleSeverity seve
     /// </returns>
     public BrokenRule GetFirstMessage(string property, RuleSeverity severity)
     {
-      return this.FirstOrDefault(c => c.Property == property && c.Severity == severity);
+      return this.OrderBy(c => c.Property).FirstOrDefault(c => c.Property == property && c.Severity == severity);
```
Comment (rockfordlhotka) [inline Source/Csla/Rules/BrokenRulesCollection.cs:286]: Shouldn't the `OrderBy` be by priority, not property?

**Review by rockfordlhotka — CHANGES_REQUESTED** [review]

**Comment by rockfordlhotka** [comment]
I added the BreakingChange tag to the original issue, as this does change the behavior of existing methods. Beyond my one request for a potential code change, I like the changes being made here.

**Comment by Freelancingonupwork** [comment]
> I added the BreakingChange tag to the original issue, as this does change the behavior of existing methods. Beyond my one request for a potential code change, I like the changes being made here.

Is this any changes required here? If not Can you please merge this PR?

**Comment by Freelancingonupwork** [comment]
Is this any changes required here? If not Can you please merge this PR?

**Comment by StefanOssendorf** [comment]
@rockfordlhotka Please approv so lgtm :)

**Review by rockfordlhotka — APPROVED** [review]

**Pull request merged**